httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/ResponseCachingPolicy.java

```diff
@@ -320,4 +354,87 @@ private boolean from1_0Origin(final HttpResponse response) {
         return HttpVersion.HTTP_1_0.equals(version);
     }
 
+    /**
+     * Calculates the freshness lifetime of a response, based on the headers in the response.
+     * <p>
+     * This method follows the algorithm for calculating the freshness lifetime described in RFC 7234, section 4.2.1.
+     * The freshness lifetime represents the time interval in seconds during which the response can be served without
+     * being considered stale. The freshness lifetime calculation takes into account the s-maxage, max-age, Expires, and
+     * Date headers as follows:
+     * <ul>
+     * <li>If the s-maxage directive is present in the Cache-Control header of the response, its value is used as the
+     * freshness lifetime for shared caches, which typically serve multiple users or clients.</li>
+     * <li>If the max-age directive is present in the Cache-Control header of the response, its value is used as the
+     * freshness lifetime for private caches, which serve a single user or client.</li>
+     * <li>If the Expires header is present in the response, its value is used as the expiration time of the response.
+     * The freshness lifetime is calculated as the difference between the expiration time and the time specified in the
+     * Date header of the response.</li>
+     * <li>If none of the above headers are present or if the calculated freshness lifetime is invalid, a default value of
+     * 5 minutes is returned.</li>
+     * </ul>
+     *
+     * <p>
+     * Note that caching is a complex topic and cache control directives may interact with each other in non-trivial ways.
+     * This method provides a basic implementation of the freshness lifetime calculation algorithm and may not be suitable
+     * for all use cases. Developers should consult the HTTP caching specifications for more information and consider
+     * implementing additional caching mechanisms as needed.
+     * </p>
+     *
+     * @param response the HTTP response for which to calculate the freshness lifetime
```

Comment:
@arturobernalg We stopped referencing specific RFCs in javadocs. Those URLs tend to change and also need to be updated every time the code base gets revised against a newer set of RFCs.